selfdrive/controls/lib/pathplanner.py

```diff
@@ -53,41 +56,48 @@ def update(self, CP, VM, CS, md, live100):
     angle_offset = live100.live100.angleOffset
     self.MP.update(v_ego, md)
 
-    # Run MPC
-    self.angle_steers_des_prev = self.angle_steers_des_mpc
+    cur_time = sec_since_boot()
+    self.angle_steers_des_prev = np.interp(cur_time, self.mpc_times, self.mpc_angles)
+
     curvature_factor = VM.curvature_factor(v_ego)
 
     l_poly = libmpc_py.ffi.new("double[4]", list(self.MP.l_poly))
     r_poly = libmpc_py.ffi.new("double[4]", list(self.MP.r_poly))
     p_poly = libmpc_py.ffi.new("double[4]", list(self.MP.p_poly))
 
-    # account for actuation delay
-    self.cur_state = calc_states_after_delay(self.cur_state, v_ego, angle_steers, curvature_factor, CP.steerRatio, CP.steerActuatorDelay)
-
-    v_ego_mpc = max(v_ego, 5.0)  # avoid mpc roughness due to low speed
-    self.libmpc.run_mpc(self.cur_state, self.mpc_solution,
-                        l_poly, r_poly, p_poly,
-                        self.MP.l_prob, self.MP.r_prob, self.MP.p_prob, curvature_factor, v_ego_mpc, self.MP.lane_width)
+    # account for actuation delay AND interpolation
+    self.cur_state = calc_states_after_delay(self.cur_state, v_ego, angle_steers, curvature_factor, CP.steerRatio, CP.steerActuatorDelay + _DT_MPC)
 
     # reset to current steer angle if not active or overriding
     if active:
-      delta_desired = self.mpc_solution[0].delta[1]
+      self.cur_state[0].delta = math.radians(self.angle_steers_des_prev - angle_offset) / CP.steerRatio
```

Comment:
this PR should be updated with the latest OP, sine steerRatio is now a learned parameter.